Software/Program/Code/Core/Inc/main.h

```diff
@@ -31,22 +31,60 @@ extern "C" {
 
 /* Private includes ----------------------------------------------------------*/
 /* USER CODE BEGIN Includes */
-
+#include <stdbool.h>
+#include <stdint.h>
```

Comment:
Does not need to be shared by everything that includes `main.h` (e.g., `main.c`, `stm32h7xx_hal_msp.c`, and `stm32h7xx_it.c`). Let the downstream decide to include these or not.